Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php

```diff
@@ -0,0 +1,782 @@
+                $data = $this->getData();
+                $data = array_merge(...array_values($data));
+                $data = $this->sortByTitle($data);
+                return ['' => $data];
+            case ilPDSelectedItemsBlockConstants::SORT_BY_START_DATE:
+                return $this->groupItemsByStartDate();
+            case ilPDSelectedItemsBlockConstants::SORT_BY_TYPE:
+                return $this->groupItemsByType();
+            case ilPDSelectedItemsBlockConstants::SORT_BY_LOCATION:
+            default:
+                return $this->groupItemsByLocation();
+        }
+    }
+
+    #[NoReturn]
+    public function removeFromDeskRoundtripObject(): void
+    {
+        $page = '';
+        if ($this->http->wrapper()->query()->has('page')) {
+            $page = $this->http->wrapper()->query()->retrieve('page', $this->refinery->kindlyTo()->string());
+        }
+
+        if ($this->http->wrapper()->query()->has('replaceSignal')) {
+            $signalId = $this->http->wrapper()->query()->retrieve('replaceSignal', $this->refinery->kindlyTo()->string());
+            $replace_signal = new ReplaceSignal($signalId);
+        }
+
+        switch ($page) {
+            case 'manage':
+                $modal = $this->ui->factory()->modal()->roundtrip(
+                    $this->getRemoveMultipleActionText(),
+                    $this->ui->factory()->legacy($this->manage($replace_signal))
+                );
+                $modal = $modal->withAdditionalOnLoadCode(function ($id) {
+                    return "
+                    $('#$id').attr('data-modal-name', 'remove_modal');
+                    ";
+                });
+                break;
+            case 'confirm':
+            default:
+                $modal = $this->ui->factory()->legacy($this->confirmRemoveObject());
+        }
+        echo $this->ui->renderer()->renderAsync($modal);
+        exit;
+    }
+
+    public function manage(ILIAS\UI\Component\ReplaceSignal $replace_signal = null): string
+    {
+        $page = '';
+        if ($this->http->wrapper()->query()->has('page')) {
+            $page = $this->http->wrapper()->query()->retrieve('page', $this->refinery->kindlyTo()->string());
+        }
+        $top_tb = new ilToolbarGUI();
+        $top_tb->setFormAction($this->ctrl->getFormAction($this, 'confirmRemove'));
+        $top_tb->setFormName('pd_remove_multiple');
+        $top_tb->setId('pd_remove_multiple');
+        $top_tb->setLeadingImage(ilUtil::getImagePath('arrow_upright.svg'), $this->lng->txt('actions'));
+        $url = $this->ctrl->getLinkTarget(
+            $this,
+            'removeFromDeskRoundtrip',
+            '',
+            true
+        ) . '&page=confirm';
+        $button = $this->ui->factory()->button()->standard($this->getRemoveMultipleActionText(), '#')
+        ->withOnLoadCode(function ($id) use ($url) {
+            return "
+            $('#$id').on('click', function() {
```

Comment:
Please move this into an own file. Replace var with const / let. Consider replacing jQuery calls to native calls if there is no obvious benefit to keeping jQuery (I will probably do a proper review when it has an own file.

Reply:
moved to own file and removed native calls